### PR TITLE
Adding telemetry buffer.

### PR DESCRIPTION
**Pull request by jaer-tsun**
**What this PR does / why we need it**:
Adding telemetry buffer.

### CODE REVIEW

File: telemetry/telemetrybuffer.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
+// Copyright 2017 Microsoft. All rights reserved.
```
Comment (ashvindeodhar) [inline telemetry/telemetrybuffer.go]: 2018

File: telemetry/telemetrybuffer.go [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+			for {
+				// Spawn worker goroutines to communicate with client
+				conn, err := tb.listener.Accept()
+				if err == nil {
+					tb.connections = append(tb.connections, conn)
+					go func() {
+						for {
+							reportStr, err := read(conn)
+							if err == nil {
+								var tmp map[string]interface{}
```
Comment (ashvindeodhar) [inline telemetry/telemetrybuffer.go]: Can you change the name of this variable from tmp to something more meaningful?

File: telemetry/telemetrybuffer.go [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+// Write - write to the file descriptor
+func (tb *TelemetryBuffer) Write(b []byte) (c int, err error) {
+	b = append(b, Delimiter)
+	w := bufio.NewWriter(tb.client)
+	c, err = w.Write(b)
+	if err == nil {
+		w.Flush()
+	}
+
+	return
```
Comment (ashvindeodhar) [inline telemetry/telemetrybuffer.go]: Isn't this return redundant?

File: telemetry/telemetrybuffer.go [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+	var body bytes.Buffer
+	json.NewEncoder(&body).Encode(tb.payload)
+	resp, err := httpc.Post(HostNetAgentURL, ContentType, &body)
+	if err != nil {
+		return fmt.Errorf("[Telemetry] HTTP Post returned error %v", err)
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
```
Comment (ashvindeodhar) [inline telemetry/telemetrybuffer.go]: replace 200 with httpOK

File: telemetry/telemetrybuffer_linux.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
+// Copyright 2017 Microsoft. All rights reserved.
```
Comment (ashvindeodhar) [inline telemetry/telemetrybuffer_linux.go]: 2018

File: telemetry/telemetrybuffer_windows.go [hunk 6]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2017 Microsoft. All rights reserved.
```
Comment (ashvindeodhar) [inline telemetry/telemetrybuffer_windows.go]: 2018

File: telemetry/telemetrybuffer.go (hunk 2 above)
Comment (jaer-tsun) [reply telemetry/telemetrybuffer.go]: will change when we discover a better name

File: telemetry/telemetrybuffer.go (hunk 3 above)
Comment (jaer-tsun) [reply telemetry/telemetrybuffer.go]: must return in golang

**Pull request merged**